### ISSUE TITLE
Provider cache - aliased providers failure

### ISSUE THREAD

**Issue by mikesplain** [gruntwork-io/terragrunt#3062]
**Describe the bug**
Thank you for the excellent work on the providers cache, our tests have shown this will make our lives SOOO much better!

We were initially blocked by #3050 but #3057 fixed that. We now see that aliased providers result in the following error.
```│ Error: Undefined provider configuration
│ 
│ The root module declares that it requires the caller to pass a
│ configuration for provider registry.terraform.io/hashicorp/aws with alias
│ "other-account".
```

**To Reproduce**
We declare our providers using terragrunt's generate command. The generic providers work fine (e.g. aws) but when using an alias, it seems like the proxy is unable to provide the required provider/alias.

```hcl
# providers.hcl
provider "aws" {
 # standard provider config
}
provider "aws" {
  alias  = "other-account"
  #standard provider config
}
```

**Expected behavior**
Expected alias provider to work and be cached.

**Nice to have**
- [ ] Terminal output
- [ ] Screenshots

**Versions**
- Terragrunt version: 0.57.1
- Terraform version: 1.7.5
- Environment details (Ubuntu 20.04, Windows 10, etc.): ubuntu 22.04

**Additional context**

**Comment by denis256** [comment]
Hi,
Could you share example code/steps where this issue happens?
I tried to reproduce a similar setup in https://github.com/denis256/terragrunt-tests/tree/master/provider-cache/alias  but mentioned error is not appearing

**Comment by levkohimins** [comment]
I couldn't repro the issue as well.

**Comment by mikesplain** [comment]
Thank you for looking! I just figured out the bug with one of our modules. Apologies for the noise.
